### PR TITLE
discord-gamesdk: install header file

### PR DESCRIPTION
**Pull request by tomodachi94**
###### Description of changes
- Split derivation into two outputs
- Install provided C header files into `$dev`
- Specify `sourceProvenance` to be a binary blob

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/discord-gamesdk/default.nix [hunk 1]
```diff
@@ -14,6 +15,10 @@ stdenv.mkDerivation rec {
     stripRoot = false;
   };
 
+  outputs = [ "out" "dev" ];
+
+  buildInputs = [ gcc.cc.libgcc ];
```
Comment (Artturin) [inline pkgs/development/libraries/discord-gamesdk/default.nix:20]:
```suggestion
  buildInputs = [ (stdenv.cc.cc.libgcc or null) ];
```
`buildInputs` should be below `nativeBuildInputs`

Comment (tomodachi94) [reply pkgs/development/libraries/discord-gamesdk/default.nix]: Done, except wdym "remember to remove the input"? Which input, where?

Comment (tomodachi94) [reply pkgs/development/libraries/discord-gamesdk/default.nix]: Done.

File: pkgs/development/libraries/discord-gamesdk/default.nix [hunk 2]
```diff
@@ -27,6 +29,8 @@ stdenv.mkDerivation rec {
       runHook preInstall
 
       install -Dm555 lib/${processor}/discord_game_sdk${sharedLibrary} $out/lib/discord_game_sdk${sharedLibrary}
+
```
Comment (Artturin) [inline pkgs/development/libraries/discord-gamesdk/default.nix:32]:
```suggestion

```

Comment (tomodachi94) [reply pkgs/development/libraries/discord-gamesdk/default.nix:32]: Fixed

**Pull request merged**